src/gpio.rs

```diff
@@ -249,14 +249,18 @@ macro_rules! gpio {
             }
 
             impl<MODE> OutputPin for $PXx<Output<MODE>> {
-                fn set_high(&mut self) {
+                type Error = ();
```

Comment:
IMO the `Error` should either be [`Infallible`](https://doc.rust-lang.org/core/convert/enum.Infallible.html), `void` or `!`, but **not** `()`.